admin-base/core/src/main/java/com/peregrine/admin/resource/AdminResourceHandlerService.java

```diff
@@ -448,18 +444,20 @@ public DeletionResponse deleteResource(ResourceResolver resourceResolver, String
             // Under these circumstances, the deletion will continue and a warning will be printed to the logs.
             logger.warn(FAILED_CREATE_RECYCLEABLE, path, e);
         }
+
         try {
-            final String primaryTypeValue = resource.getValueMap().get(JCR_PRIMARY_TYPE, EMPTY);
+            final String primaryTypeValue = PerUtil.getPrimaryType(resource);
             if (isNotEmpty(primaryType) && !primaryTypeValue.equals(primaryType)) {
                 throw new ManagementException(String.format(PRIMARY_TYPE_ASKEW_FOR_DELETION, path, primaryType, primaryTypeValue));
             }
 
-            final Resource parent = resource.getParent();
+            final Optional<Resource> parent = Optional.ofNullable(resource.getParent());
```

Comment:
does this refactor improve the code from a memory footprint perspective?

Reply:
Code gets more linear, the number of branches in the flow is reduced below.